### PR TITLE
NCL-4587 Add injection for automatic usage of the ManipulationPlugin

### PR DESCRIPTION
**Pull request by rnc**
This is in draft as its waiting on #26 by @TomasHofman  to be completed.

### CODE REVIEW

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java [hunk 1]
```diff
@@ -58,7 +67,45 @@ public void perform() {
 
         projectsToAlign.remove(projectName);
         if (projectsToAlign.isEmpty()) {
+
+            System.out.println("#### settings is " + project.getGradle().getRootProject() + " and "
```
Comment (metacosm) [inline analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java]: Should we keep this output?

File: analyzer/src/main/resources/gme.gradle [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    dependencies {
+        classpath "org.jboss.gm.manipulation:manipulation:${project.version}"
+    }
+}
+
+// If we are not running the analyse plugin then apply the manipulation plugin.
+// This avoids issues with rerunning in local environments.
+if (!project.hasProperty("gmeAnalyse")) {
```
Comment (metacosm) [inline analyzer/src/main/resources/gme.gradle]: Where does that come into play? I don't see any reference to this property anywhere (though I might just have missed it)…

File: analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java (hunk 1 above)
Comment (rnc) [reply analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java]: Damn. Left over debugging! I'll remove.

File: analyzer/src/main/resources/gme.gradle (hunk 2 above)
Comment (rnc) [reply analyzer/src/main/resources/gme.gradle]: Its linked to my init.gradle which sets the value to true. 
I'll create a separate ticket for this as I think we should store the init.gradle in the repo somewhere so we can discuss and have things like
```
allprojects {
    ext.gmeAnalyse = true
    apply plugin: org.jboss.gm.analyzer.alignment.AlignmentPlugin
```

Comment (metacosm) [reply analyzer/src/main/resources/gme.gradle]: I agree about storing the `init.gradle` in the repo: I had actually started working on it at some point.

**Pull request merged**